src/api/users/src/routes/users.js

```diff
@@ -28,29 +28,50 @@ router.get('/:id', validateId(), async (req, res, next) => {
 });
 
 // get all users
-// rejected if the user collection is empty, otherwise users returned
+// rejected if the users collection is empty
+// otherwise n (specified via params) users are returned
 router.get('/', async (req, res, next) => {
-  try {
-    const usersRef = db.collection('users');
-    const snapshot = await usersRef.get();
+  const page = !parseInt(req.query.page, 10) ? 1 : parseInt(req.query.page, 10); // if req.query.page is 0 or unparsable set to 1, otherwise set to the req.query.page
+  const perPage = !parseInt(req.query.per_page, 10) ? 5 : parseInt(req.query.per_page, 10);
```

Comment:
Let's use 10 or 20 as our default vs. 5.

Reply:
Also, use `Math.min()` here so that we only allow getting values between 10 and 100 (or some other upper-bound).  Don't let the user define some massive number.  For example, [GitHub's Users API](https://docs.github.com/en/rest/reference/users#list-users) sets an upper limit of 100.

Reply:
> Also, use Math.min() here

Love it!! Thanks!